implementations/micrometer-registry-prometheus/src/main/java/io/micrometer/prometheus/PrometheusLongTaskTimer.java

```diff
@@ -16,53 +16,21 @@
 package io.micrometer.prometheus;
 
 import io.micrometer.core.instrument.Clock;
-import io.micrometer.core.instrument.distribution.CountAtBucket;
 import io.micrometer.core.instrument.distribution.DistributionStatisticConfig;
-import io.micrometer.core.instrument.distribution.HistogramSnapshot;
-import io.micrometer.core.instrument.internal.DefaultLongTaskTimer;
-import io.micrometer.core.lang.Nullable;
+import io.micrometer.core.instrument.internal.CumulativeHistogramLongTaskTimer;
 
-import java.util.Arrays;
 import java.util.concurrent.TimeUnit;
-import java.util.concurrent.atomic.AtomicInteger;
 
 /**
  * Extends the default long task timer, making histogram counts cumulative over time.
  *
  * @author Jon Schneider
  * @since 1.5.0
+ * @deprecated since 1.5.2, use {@link CumulativeHistogramLongTaskTimer}.
  */
-public class PrometheusLongTaskTimer extends DefaultLongTaskTimer {
-    @Nullable
-    private CountAtBucket[] lastSnapshot;
-
+@Deprecated
```

Comment:
Do you think these can be considered to be internal implementation details not subject to the same compatibility guarantees? Is there any scenario in which someone could break unless they explicitly casted the `LongTaskTimer` returned by `PrometheusMeterRegistry` to this type?

Reply:
@jkschneider I just did this as it is `public`. Actually, I was wondering if registry-specifc meter implementations need to be `public` as some of them are `public`, but the others are package-private. If it's safe to assume that it's internal, I'll drop it completely. Let me know what you think.

Reply:
@izeye Yes, I think they should have all been package private really.

Reply:
@jkschneider Thanks for the feedback! I removed it.